lib/pbench/test/unit/server/test_relay.py

```diff
@@ -0,0 +1,318 @@
+        }
+
+    @responses.activate
+    def test_relay_no_json(self, client, server_config, pbench_drb_token, tarball):
+        file, md5file, md5 = tarball
+        responses.add(
+            responses.GET, "https://relay.example.com/uri1", status=HTTPStatus.NOT_FOUND
+        )
+        responses.add(
+            responses.GET, "https://relay.example.com/uri2", status=HTTPStatus.NOT_FOUND
+        )
+        response = client.post(
+            self.gen_uri(server_config, "https://relay.example.com/uri1"),
+            headers=self.gen_headers(pbench_drb_token),
+        )
+        assert (
+            response.status_code == HTTPStatus.BAD_GATEWAY
+        ), f"Unexpected result, {response.text}"
+
+    @responses.activate
+    def test_relay_not_json(self, client, server_config, pbench_drb_token):
+        responses.add(
+            responses.GET,
+            "https://relay.example.com/uri1",
+            status=HTTPStatus.OK,
+            body="This isn't JSON",
+        )
+        responses.add(
+            responses.GET, "https://relay.example.com/uri2", status=HTTPStatus.NOT_FOUND
+        )
+        response = client.post(
+            self.gen_uri(server_config, "https://relay.example.com/uri1"),
+            headers=self.gen_headers(pbench_drb_token),
+        )
+        assert (
+            response.status_code == HTTPStatus.BAD_GATEWAY
+        ), f"Unexpected result, {response.text}"
+
+    @responses.activate
+    def test_relay_missing_json_field(
+        self, client, server_config, pbench_drb_token, tarball
+    ):
+        file, md5file, md5 = tarball
+        responses.add(
+            responses.GET,
+            "https://relay.example.com/uri1",
+            status=HTTPStatus.OK,
+            json={
+                "name": "tarball.tar.xz",
+                "md5": md5,
+                "access": "private",
+                "metadata": [],
+            },
+        )
+        responses.add(
+            responses.GET, "https://relay.example.com/uri2", status=HTTPStatus.NOT_FOUND
+        )
+        response = client.post(
+            self.gen_uri(server_config, "https://relay.example.com/uri1"),
+            headers=self.gen_headers(pbench_drb_token),
+        )
+        assert (
+            response.status_code == HTTPStatus.BAD_GATEWAY
+        ), f"Unexpected result, {response.text}"
+
+    def test_relay_bad_prepare(
+        self, client, server_config, pbench_drb_token, monkeypatch
+    ):
+        def throw(self, args: ApiParams, request: Request) -> Intake:
+            raise Exception("An exception that's not APIAbort")
+
+        monkeypatch.setattr(Relay, "_prepare", throw)
+        response = client.post(
+            self.gen_uri(server_config, "https://relay.example.com/uri1"),
+            headers=self.gen_headers(pbench_drb_token),
+        )
+        assert (
+            response.status_code == HTTPStatus.INTERNAL_SERVER_ERROR
+        ), f"Unexpected result, {response.text}"
```

Comment:
This isn't really a test of anything in `pbench.server.api.resources.relay`...should it live somewhere else?  (We don't actually have a unit test module for the base class, do we?  🤔)

Ditto for `test_relay_bad_access()`.